azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/implementation/NetworkImpl.java

```diff
@@ -112,17 +108,17 @@ public NetworkImpl withSubnets(Map<String, String> nameCidrPairs) {
     public NetworkImpl withoutSubnet(String name) {
         // Remove from cache
         List<Subnet> s = this.subnets;
-        for(int i=0; i<s.size(); i++) {
-            if(s.get(i).name().equalsIgnoreCase(name)) {
+        for (int i = 0; i < s.size(); i++) {
+            if (s.get(i).name().equalsIgnoreCase(name)) {
                 s.remove(i);
                 break;
             }
         }
 
         // Remove from inner
         List<SubnetInner> innerSubnets = this.inner().subnets();
-        for(int i=0; i<innerSubnets.size(); i++) {
-            if(innerSubnets.get(i).name().equalsIgnoreCase(name)) {
+        for (int i = 0; i < innerSubnets.size(); i++) {
+            if (innerSubnets.get(i).name().equalsIgnoreCase(name)) {
                 innerSubnets.remove(i);
                 break;
             }
```

Comment:
when we separate the logic of removing from the "truth" (inner) from the logic of removing from the cache, I worry it's an open invitation for bugs where the cache is not in synch with the "truth"...